locations/spiders/vr_bank.py

```diff
@@ -0,0 +1,188 @@
+    'Samstag': 'Sa',
+    'Sonntag': 'Su'
+}
+
+
+class VRBankSpider(scrapy.Spider):
+    name = "vr_bank"
+    allowed_domains = ["www.vr.de"]
+    start_urls = ['https://www.vr.de/service/filialen-a-z/a.html']
+
+    def process_hours(self, store_hours):
+        opening_hours = OpeningHours()
+
+        day = time = open_time = close_time = ""
+
+        if store_hours is None:
+            return
+
+        for hour in store_hours:
+            hour = hour.replace(": ", "#")
+            try:
+                day, time = hour.split("#")
+            except Exception:
+                pass
+
+            if time and day in DAY_MAPPING.keys():
+                time = time.replace("24:00", "00:00")
+                tms = []
+                if 'und' in time:
+                    tms = [tm.strip() for tm in time.split('und')]
+                else:
+                    tms.append(time)
+
+                for tm in tms:
+                    try:
+                        open_time, close_time = [t.strip() for t in tm.replace('Uhr', '').strip().split("-")]
+                    except Exception:
+                        pass
+
+                    if open_time and close_time and day:
+                        opening_hours.add_range(
+                            day=DAY_MAPPING[day],
+                            open_time=open_time,
+                            close_time=close_time,
+                            time_format='%H:%M'
+                        )
+                    else:
+                        continue
+
+        return opening_hours.as_opening_hours()
+
+    def parse_details(self, response):
+        if response.status == 403:
+            yield scrapy.Request(
+                url=response.meta.get('url'),
+                callback=self.parse_details,
+                meta={'url': response.meta.get('url')}
+            )
+
+        name = street = zip = city = phone = website = latitude = longitude = ""
+
+        try:
+            name = response.xpath('//h1[@itemprop="name"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            street = response.xpath('//span[@itemprop="streetAddress"]/text()').get()
+        except Exception:
+            pass
+        try:
+            zip = response.xpath('//span[@itemprop="postalCode"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            city = response.xpath('//span[@itemprop="addressLocality"]/text()').get()
+        except Exception:
+            pass
+
+        try:
+            phone = response.xpath('//li[@itemprop="telephone"]/a/span/text()').get()
+        except Exception:
+            pass
+
+        try:
+            website = response.xpath('//li[@itemprop="url"]/a/span/text()').get()
+        except Exception:
+            pass
+
+        try:
+            m = re.search(r'lat&quot;:([-+]?[0-9]*\.?[0-9]*)', response.text)
+            if m:
+                latitude = m.group(1)
+        except Exception:
```

Comment:
Please don't wrap all these in exception catches. If something fails, we want to fix it.

Reply:
Actually these are expected to fail. Like some banks do not provide working hours. In that case the regex will fail.

Reply:
Please add a comment and use a more restrictive exception in the `except` block so we don't swallow all exceptions.

Reply:
Please check now.

Reply:
What I meant by "use a more restrictive exception" is to figure out which exception is thrown that you want to catch and catch only that one instead of catching `Exception`.

Reply:
Done. I have removed unwanted exception handling and elsewhere made it more restrictive to ValueError.